inst/www/htmlwidgets.js

```diff
@@ -233,7 +233,21 @@
           theseArgs = theseArgs.concat([task.data]);
           task = task.code;
         }
-        var taskFunc = eval("(" + task + ")");
+        var taskFunc = null;
+        try {
+          taskFunc = eval(task);
+        } catch(error) {
+          if (!error instanceof SyntaxError) {
+            throw new Error(error.message);
+          }
+          // To support function declarations, retry evaluation with
+          // parentheses (turns declarations into an expression)
+          try {
+            taskFunc = eval("(" + task + ")");
+          } catch(e) {
+            throw new Error(error.message);
```

Comment:
If and only if `e instanceof SyntaxError` do we `throw error;`. Otherwise, we `throw e;`.